app/src/common/shared/org/mozilla/vrbrowser/SessionStore.java

```diff
@@ -416,14 +416,18 @@ public void setCurrentSession(int aId) {
 
     public void setRegion(String aRegion) {
         Log.d(LOGTAG, "SessionStore setRegion: " + aRegion);
-        aRegion = aRegion != null ? aRegion.toLowerCase() : "worldwide";
-        mRegion = aRegion;
+        mRegion = aRegion != null ? aRegion.toLowerCase() : "worldwide";
+
+        // There is a region update and the home is already loaded
+        if (mCurrentSession != null && isHomeUri(getCurrentUri())) {
+            mCurrentSession.loadUri(SessionStore.HOME_WITHOUT_REGION_ORIGIN + "#{'region':'" + mRegion + "'}");
```

Comment:
feel free to change this to `#region=mRegion`; I have code in place for parsing `#history=` already, so it'll be easy to accept